docs/source/guides/mod-config.md

```diff
@@ -1,166 +1,328 @@
+			Version 1.0.0\n\z
+			\n\z
+			The mod is very awesome. And this is its awesome description.\n",
 	})
 
 	settings:createYesNoButton({
 		label = "Enable Mod",
+		configKey = "enabled",
 		-- This description will be displayed at the right panel if the button is moused over.
 		description =
-			"If this setting is enabled, the mod is enabled.\n" ..
-			"\n" ..
-			"If this setting is disabled, the mod is disabled.\n" ..
-			"\n" ..
-			"Default: yes",
-		variable = mwse.mcm:createTableVariable({ id = "enabled", table = config }),
+			"If this setting is enabled, the mod is enabled.\n\z
+			\n\z
+			If this setting is disabled, the mod is disabled."
 	})
-	```
+```
 
-### Key Binder
+### Displaying Default Values of Settings
+As your config menu grows, you may want to include some information about what the default values for various settings are. 
+Fortunately, the MCM comes with a very straight-forward solution to this problem:
+```lua hl_lines="4 5"
+local template = mwse.mcm.createTemplate({ 
+	name = "My Awesome Mod", 
+	config = myConfig,
+	defaultConfig = myDefaultConfig,
+	showDefaultSetting = true,
+})
+```
 
-If your mod adds any hotkey function, a way to rebind the hotkey in the mod config menu is a must.
+This will result in the default values of settings automatically being displayed underneath their descriptions, provided those settings are in a `SideBarPage`.
 
-!!! example "**MyMod\\main.lua**"
-	```lua
-	settings:createKeyBinder({
-		label = "Assign Keybind",
-		description = "Assign a new keybind to perform awesome tasks.",
-		variable = mwse.mcm.createTableVariable{ id = "keybind", table = config },
-		allowCombinations = true,
-	})
-	```
+### Separate config.lua
+
+As your default config gets long, you might want to create a separate Lua file for it.
+
+```lua linenums="1" title="My Awesome Mod/defaultConfig.lua"
+-- Define the defaultConfig in a separate file so we can access it later.
+local defaultConfig = {
+	enabled = true,
+	keybind = {
+		keyCode = tes3.scanCode.k,
+		isShiftDown = false,
+		isAltDown = false,
+		isControlDown = true,
+	}, -- Ctrl + K
+	nonEdible = {
+		["ingred_diamond_01"] = true,
+		["ingred_scrap_metal_01"] = true,
+	}
+}
+return defaultConfig
+```
+
+```lua linenums="1" title="My Awesome Mod/config.lua"
+local defaultConfig = require("My Awesome Mod.defaultConfig")
+local config = mwse.loadConfig("My Awesome Mod", defaultConfig)
+return config
+```
+
+We can then use the `require` function to import the `config` and `defaultConfig` in `main.lua`.
+
+```lua linenums="1" title="My Awesome Mod/main.lua"
+local config = require("My Awesome Mod.config")
+local defaultConfig = require("My Awesome Mod.defaultConfig")
+```
+
+!!! note
+	Having a dedicated config file is important when your mod has multiple files.
+	When using a dedicated config file, you can import it from every file of your mod using `require`, and all of those files
+	will use a sychronized copy of your config.
```

Comment:
Was it intended to have this broken into multiple lines? Markdown should insert newlines based on the resolution of the browser.

Reply:
I was not intending to break it up into multiple lines. I thought it was like `LaTeX` in the sense that you can add a single new-line for the purposes of formatting the source code and it wouldn't affect the "compiled" version. I just find the less wide lines a bit easier to read. The version I built locally doesn't appear to have any displayed new lines so I figured it would be okay. But I can change it if you would prefer.

Do the new lines affect the rendering? (If they don't, I can still change it so it's more consistent with the rest of the documentation code if you'd prefer.)

Reply:
Oh, you are totally right 😅, it doesn't affect the rendered output, also I agree about readability. I have misremembered something. We are good to go here.